### PR TITLE
[tests] use Awaitility replace Thread.sleep for pulsar-broker.

### PR DESCRIPTION
**Pull request by yangl**
### Motivation

Currently the testcase use  some Thead.sleep in the pulsar-broker module.

### Modifications

- use Awaitility replace Thread.sleep
- remove some unnecessary Thread.sleep

### CODE REVIEW

**Comment by Anonymitaet** [comment]
Thanks for your contribution. For this PR, do we need to update docs?

**Comment by yangl** [comment]
> Thanks for your contribution. For this PR, do we need to update docs?

no.

**Review by sijie — APPROVED** [review]

**Review by merlimat — APPROVED** [review]

**Review by eolivelli — CHANGES_REQUESTED** [review]
awesome work.

I let one comment and a couple of nits.

please take a look

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/namespace/OwnershipCacheTest.java [hunk 1]
```diff
@@ -355,9 +356,9 @@ public void testRemoveOwnership() throws Exception {
         assertEquals(cache.getOwnedBundles().size(), 1);
         cache.removeOwnership(bundle);
         Thread.sleep(500);
-        assertTrue(cache.getOwnedBundles().isEmpty());
+        Awaitility.await().untilAsserted(() -> assertTrue(cache.getOwnedBundles().isEmpty()));
+
 
-        Thread.sleep(500);
```
[the remark this answers is not in the document]
Comment (yangl) [reply pulsar-broker/src/test/java/org/apache/pulsar/broker/namespace/OwnershipCacheTest.java:360]: ohh, thanks.

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/service/SubscriptionSeekTest.java [hunk 2]
```diff
@@ -531,7 +532,7 @@ public void testOnlyCloseActiveConsumerForSingleActiveConsumerDispatcherWhenSeek
         assertEquals(connectedSinceSet.size(), 2);
         consumer1.seek(MessageId.earliest);
         // Wait for consumer to reconnect
-        Thread.sleep(1000);
+        Awaitility.await().until(() -> consumer1.isConnected() == true);
```
[the remark this answers is not in the document]
Comment (yangl) [reply pulsar-broker/src/test/java/org/apache/pulsar/broker/service/SubscriptionSeekTest.java]: Thank you for the review suggestion, I have fixed these, please take a look again!

**Comment by yangl** [comment]
> awesome work.
> 
> I let one comment and a couple of nits.
> 
> please take a look

Thank you for the review suggestion, I have fixed these, please take a look again!

**Review by eolivelli — APPROVED** [review]
LGTM

**Comment by eolivelli** [comment]
committed to master branch, thank you @yangl

**Pull request merged**